content/docs/tutorials/instrumenting_http_server_in_go.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Instrumenting HTTP server written in Go  
+sort_rank: 3
+---
+
+In this tutorial we will create a simple Golang HTTP server and instrumentation it by adding a counter
+metric to keep count of the total number of requests processed by the server.
+
+Here we have a simple HTTP server with `/ping` endpoint which returns `pong` as response.
+
+```go
+package main
+
+import (
+   "fmt"
+   "net/http"
+)
+
+func ping(w http.ResponseWriter, req *http.Request){
+   fmt.Fprintf(w,"pong")
+}
+
+func main() {
+   http.HandleFunc("/ping",ping)
+
+   http.ListenAndServe(":8090", nil)
+}
+```
+
+Compile and run the server
+
+```bash
+go build server.go
+./server.go
+```
+
+Now open `http://localhost:8090/ping` in your browser and you must see `pong`.
+
+[![Server](/assets/tutorial/server.png)](/assets/tutorial/server.png)
+
+
+Now lets add a metric to the server which will instrument the number of requests made to the ping endpoint,the counter metric type is suitable for this as we know the request count doesn’t go down and only increases.
+
+Create a prometheus counter
+
+```go
```

Comment:
Mention the new import you have to do for this? They are shown at the end, but should be mentioned here. Just "Note that we need to import new libraries, see the full code example below" or something.

Reply:
Updated the commands at the end to use `go modules`, therefore we don't need to get the dependencies separately.